support-modules/acquisition-events/src/main/scala/com/gu/support/acquisitions/ga/GoogleAnalyticsServiceImpl.scala

```diff
@@ -0,0 +1,255 @@
+package com.gu.support.acquisitions.ga
+
+import cats.data.EitherT
+import com.gu.acquisitionsValueCalculatorClient.model.{AcquisitionModel, PrintOptionsModel}
+import com.gu.acquisitionsValueCalculatorClient.service.AnnualisedValueService
+import com.gu.support.acquisitions.ga.GoogleAnalyticsServiceImpl._
+import com.gu.support.acquisitions.ga.models.GAError.{BuildError, NetworkFailure, ResponseUnsuccessful}
+import com.gu.support.acquisitions.ga.models.{ConversionCategory, GAData, GAError}
+import com.gu.support.acquisitions.models.AcquisitionProduct.{Contribution, DigitalSubscription, Paper, RecurringContribution}
+import com.gu.support.acquisitions.models.{AbTest, AcquisitionDataRow, AcquisitionProduct, PrintOptions}
+import com.gu.support.acquisitions.models.PrintProduct._
+import com.gu.support.acquisitions.utils.Retry
+import com.typesafe.scalalogging.LazyLogging
+import okhttp3.{Call, Callback, HttpUrl, OkHttpClient, Request, RequestBody, Response}
+
+import java.io.IOException
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.control.NonFatal
+import scala.util.matching.Regex
+
+object GoogleAnalyticsServiceImpl {
+  val clientIdPattern: Regex = raw"GA\d\.\d\.(\d+\.\d+)".r
+
+  // e.g. GUARDIAN_WEEKLY becomes GuardianWeekly
+  def camelCase(s: String): String = s
+    .split("_")
+    .map(_.toLowerCase.capitalize)
+    .mkString("")
+}
+
+trait GoogleAnalyticsService {
+  def submit(acquisition: AcquisitionDataRow, gaData: GAData, maxRetries: Int)(implicit ec: ExecutionContext): EitherT[Future, List[GAError], Unit]
+}
+
+object MockGoogleAnalyticsService extends GoogleAnalyticsService {
+  def submit(acquisition: AcquisitionDataRow, gaData: GAData, maxRetries: Int)(implicit ec: ExecutionContext): EitherT[Future, List[GAError], Unit] =
+    EitherT.fromEither(Right(()))
+}
+
+class GoogleAnalyticsServiceImpl(client: OkHttpClient) extends GoogleAnalyticsService with LazyLogging {
+  private[ga] val gaPropertyId: String = "UA-51507017-5"
+  private[ga] val endpoint: HttpUrl = HttpUrl.parse("https://www.google-analytics.com")
+
+  private[ga] def buildBody(acquisition: AcquisitionDataRow, gaData: GAData)(implicit ec: ExecutionContext): EitherT[Future, BuildError, RequestBody] = EitherT {
+    getAnnualisedValue(acquisition)
+      .fold(
+        error => {
+          logger.warn(s"Couldn't retrieve annualised value for this acquisition: $error")
+          buildPayload(acquisition, 0, gaData) // We still want to record the acquisition even if we can't get the AV
+        },
+        value => buildPayload(acquisition, value, gaData)
+      )
+      .map { maybePayload =>
+        maybePayload.map { payload =>
```

Comment:
for clarity, there's no need to repeat the buildPayload call
I'm not totally sure of the types here, but something like this would be equivalent
```suggestion
      .fold(
        error => {
          logger.warn(s"Couldn't retrieve annualised value for this acquisition: $error")
         0 // We still want to record the acquisition even if we can't get the AV
        },
        value => value
      )
      .map { maybeValue =>
        buildPayload(acquisition,maybeValue, gaData).map { payload =>
```

Reply:
It's an odd one because `getAnnualisedValue` returns an `EitherT`, `fold` gives us just a `Future`, then we end up wrapping the result in an `EitherT` anyway!
I'm just having a play...

Reply:
I've pushed this up

Reply:
looks good. 👍